backend/layers/business/business.py

```diff
@@ -141,6 +143,34 @@ def get_published_collection_version(self, collection_id: CollectionId) -> Optio
         """
         return self.database_provider.get_collection_mapped_version(collection_id)
 
+    def get_published_collection_version__discover_api(
+        self, collection_version_id: str
+    ) -> CollectionVersionWithPublishedDatasets:
+        collection_version = self.database_provider.get_collection_version_with_datasets(
+            CollectionVersionId(collection_version_id)
+        )
+        if collection_version is None or collection_version.published_at is None:
```

Comment:
It's cleaner if we keep as much http error handle in the API layer. You can make new http exception to handle these.